libvncserver/tight.c

```diff
@@ -1577,15 +1596,18 @@ SendJpegRect(rfbClientPtr cl, int x, int y, int w, int h, int quality)
         }
     }
 
-    if (tightAfterBufSize < TJBUFSIZE(w, h)) {
+    if (!tightAfterBuf || tightAfterBufSize < TJBUFSIZE(w, h)) {
         if (tightAfterBuf == NULL)
             tightAfterBuf = (char *)malloc(TJBUFSIZE(w, h));
-        else
-            tightAfterBuf = (char *)realloc(tightAfterBuf,
-                                            TJBUFSIZE(w, h));
-        if (!tightAfterBuf) {
-            rfbLog("Memory allocation failure!\n");
-            return 0;
+        else {
+            char *reallocedBeforeEncBuf = (char *)realloc(tightAfterBuf, TJBUFSIZE(w, h));
```

Comment:
same here, probbaly a c&p remnant

Reply:
Thanks for reviewing carefully - indeed just a c&p mistake which is now fixed.